### ISSUE TITLE
false positive on no-whitespace-for-layout

### ISSUE THREAD

**Issue by nvdk** [ember-template-lint/ember-template-lint#2899]
I'm getting a linting error on the following code (running ember-template-lint 5.10.0):

```
                <button type="button"
                  {{on "click" (fn @onChange zeroBasedPage)}}
                  class="-ml-px relative inline-flex items-center {{this.paddingClass}} border border-zinc-300 bg-white {{this.textSizeClass}} leading-5 font-medium {{if (eq zeroBasedPage this.page) "text-blue-600 hover:text-blue-800 active:text-blue-600" "text-zinc-700 hover:text-zinc-500 active:text-zinc-700"}} focus:z-10  focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500">
                  {{page}}
                </button>
```
it seems to be angry about the whitespace after the inline handlebars in the class attribute (column 142 of that line). I know this isn't the prettiest code, but shouldn't the rule ignore attributes?

**Comment by bmish** [comment]
Currently, as the [rule doc](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-whitespace-for-layout.md) says, this rule checks any text nodes for excess whitespace. But I think it's a fair question to ask if the rule should be checking the values of arbitrary attributes. Arguably, the `class` attribute could be fair-game for an excess whitespace check, as it's an attribute we know shouldn't have (or at least doesn't need) excess whitespace in it, but attribute values aren't used for the page layout, so checking this may not align with the goal of the rule.

I didn't immediately see any justification for this behavior in the original rule PR #819. Maybe @MelSumner has some context.

**Comment by MelSumner** [comment]
@bmish I'd be okay with excluding attribute values from these checks, that seems reasonable.

**Comment by lifeart** [comment]
Closing issue, but it's still listed here: https://github.com/ember-template-lint/ember-template-lint/issues/3209
